Gems/ROS2/Code/Source/Transform/ROS2FrameComponent.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "TransformPublisher.h"
+#include <AzCore/Component/Component.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+
+namespace ROS2
+{
+    /// This component marks an interesting reference frame and captures common behavior
+    /// for sensor data frame of reference, publishing ros2 static and dynamic transforms (tf)
+    /// and helping to export robot prefab to URDF.
+    class ROS2FrameComponent
+        : public AZ::Component
+    {
+    public:
+        AZ_COMPONENT(ROS2SensorComponent, "{EE743472-3E25-41EA-961B-14096AC1D66F}", AZ::Component);
+
+        void Activate() override;
+        void Deactivate() override;
+
+        static void Reflect(AZ::ReflectContext* context);
+        static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided);
+        static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible);
+        static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
+
+        // namespaced and validated, ready to send in ros2 messages (TODO-validate)
+        AZStd::string GetFrameID() const;
+        AZStd::string GetNamespace() const;
+
+        // Transform for direct parent if parent ROS2Frame is found, otherwise transformation is global ("world")
+        const AZ::Transform& GetFrameTransform() const;
+
+    private:
+        AZ::TransformInterface* GetEntityTransformInterface() const;
+        const ROS2FrameComponent* GetParentROS2FrameComponent() const;
```

Comment:
I think this function might be useful not only for the class, so perhaps we can make it public?

Reply:
I guess it returns null when the parent frame is "world". What do you think about having a ROS2FrameComponent singleton representing "world" and returning its pointer here? I think it would be a more convenient interface.

Reply:
Unsure about this. I am not sure what would be the use case to get parent frame outside of what is already directly exposed in public interface (e.g. give me a transform). If I get a good example (on the way) I might reconsider, by default I prefer to keep things... private ;)

As for specific component, I am not a fan of it. "World" is just a string with ros2 semantics, it would not make much sense with current API for the component. I don't like nullptr logic at all, but it seems that Entities and Components do embrace it. Still open to ideas.

Reply:
Discussed F2F. World component would probably not do things that other components do and it's not clear what object should have this component.